tests/bot/cogs/test_antimalware.py

```diff
@@ -0,0 +1,146 @@
+        self.message.attachments = [attachment]
+        self.message.guild = None
+
+        await self.cog.on_message(self.message)
+
+        self.message.delete.assert_not_called()
+
+    async def test_message_with_illegal_extension_gets_deleted(self):
+        """A message containing an illegal extension should send an embed."""
+        attachment = MockAttachment(filename="python.asdfsff")
+        self.message.attachments = [attachment]
+
+        await self.cog.on_message(self.message)
+
+        self.message.delete.assert_called_once()
+
+    async def test_message_send_by_staff(self):
+        """A message send by a member of staff should be ignored."""
+        moderator_role = MockRole(name="Moderator", id=Roles.moderators)
+        self.message.author.roles.append(moderator_role)
+        attachment = MockAttachment(filename="python.asdfsff")
+        self.message.attachments = [attachment]
+
+        await self.cog.on_message(self.message)
+
+        self.message.delete.assert_not_called()
+
+    async def test_python_file_redirect_embed(self):
+        """A message containing a .python file should result in an embed redirecting the user to our paste site"""
+        attachment = MockAttachment(filename="python.py")
+        self.message.attachments = [attachment]
+        self.message.channel.send = AsyncMock()
+
+        await self.cog.on_message(self.message)
+        args, kwargs = self.message.channel.send.call_args
+        embed = kwargs.pop("embed")
+
+        self.assertEqual(args[0], f"Hey {self.message.author.mention}!")
+        self.assertEqual(embed.description, (
+            "It looks like you tried to attach a Python file - "
+            f"please use a code-pasting service such as {URLs.site_schema}{URLs.site_paste}"
+        ))
+
+    async def test_txt_file_redirect_embed(self):
+        attachment = MockAttachment(filename="python.txt")
+        self.message.attachments = [attachment]
+        self.message.channel.send = AsyncMock()
+
+        await self.cog.on_message(self.message)
+        args, kwargs = self.message.channel.send.call_args
+        embed = kwargs.pop("embed")
+        cmd_channel = self.bot.get_channel(Channels.bot_commands)
+
+        self.assertEqual(args[0], f"Hey {self.message.author.mention}!")
+        self.assertEqual(embed.description, (
+            "**Uh-oh!** It looks like your message got zapped by our spam filter. "
+            "We currently don't allow `.txt` attachments, so here are some tips to help you travel safely: \n\n"
+            "• If you attempted to send a message longer than 2000 characters, try shortening your message "
+            "to fit within the character limit or use a pasting service (see below) \n\n"
+            "• If you tried to show someone your code, you can use codeblocks \n(run `!code-blocks` in "
+            f"{cmd_channel.mention} for more information) or use a pasting service like: "
+            f"\n\n{URLs.site_schema}{URLs.site_paste}"
```

Comment:
I've been looking into this but since two of the strings use variables that aren't available at that point I was thinking of moving the embed creating to their own functions and mocking the return value of those.